File: algorithm/python/BoatsToSavePeople.py
class Solution:
    def numRescueBoats(self, people, limit):
        res = 0
        people.sort()
        light, heavy = 0, len(people) - 1

        while light <= heavy:
            if people[light] + people[heavy] > limit:
                heavy -= 1
            else:
                light += 1
                heavy -= 1
            res += 1
        
        return res

File: algorithm/JavaScript/BoatsToSavePeople.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
  let res = 0;
  people.sort((a, b) => a - b)
  let light = 0, heavy = people.length - 1
  while (light <= heavy) {
      if (people[light] + people[heavy] > limit) {
          heavy -= 1
      } else {
          heavy -= 1
          light += 1
      }
      res += 1
  }
  return res
};